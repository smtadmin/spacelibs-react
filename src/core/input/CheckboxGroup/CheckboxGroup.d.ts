export default CheckboxGroup;
declare class CheckboxGroup {
    constructor(props: object);
    state: {
        values: any[];
    };
    onValueChanged(identifier: object): void;
    render(): object;
}
declare namespace CheckboxGroup {
    namespace defaultProps {
        const color: string;
        const alignment: string;
        const labelPlacement: string;
    }
    namespace propTypes {
        export const value: any;
        export const hasError: any;
        export const subText: any;
        const alignment_1: any;
        export { alignment_1 as alignment };
        const color_1: any;
        export { color_1 as color };
        const labelPlacement_1: any;
        export { labelPlacement_1 as labelPlacement };
        export const variant: any;
        export const inputLabel: any;
        export const placeholder: any;
        export const onValueChanged: any;
        export const hideSelectedValueInInput: any;
        export const config: any;
    }
}
