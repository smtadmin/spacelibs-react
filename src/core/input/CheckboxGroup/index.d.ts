export default CheckboxGroup;
import CheckboxGroup from "./CheckboxGroup";
