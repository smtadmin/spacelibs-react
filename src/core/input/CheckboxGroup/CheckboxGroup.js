/*
 * File: /src/core/input/CheckboxGroup/CheckboxGroup.js
 * Version: 0.0.2
 * Project: @siliconmtn/spacelibs-react
 * Description: Checkbox Field input
 * File Created: Monday, 8th February 2021 1:27 pm
 * Author: tyler Gaffaney (tyler.gaffaney@siliconmtn.com)
 * -----
 * Last Modified: Wednesday, 3rd March 2021 5:03 pm
 * Modified By: Justin Jeffrey (justin.jeffrey@siliconmtn.com>)
 * -----
 * Copyright 2021, Silicon Mountain Technologies, Inc.
*/

import React from 'react';
import PropTypes from 'prop-types';import FormGroup from '@material-ui/core/FormGroup';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Checkbox from '@material-ui/core/Checkbox';
import Tooltip from '../Tooltip';

/**
 * Class for Checkbox buttons
 */
class CheckboxGroup extends React.Component {	
	
	/**
	 * Creates an instance of CheckboxGroup.
	 * @param {object} props - Component props
	 * @memberof CheckboxGroup
	 */
	constructor(props){
		super(props);		
		var selectedValues = [];
		for(var y = 0; y < props.value.length; y++){
			var val = props.value[y];
			selectedValues.push(val.identifier);
		}		

		var valueObjects = [];
		for(var x = 0; x < props.config.options.length; x++){
			var value = props.config.options[x];
			
			if(value.isSelected === undefined){
				value.isSelected = false;
			}
			
			if(value.isDisabled === undefined){
				value.isDisabled = false;
			}			
			
			if(selectedValues.includes(value.identifier)){
				value.isSelected = true;
			}
			valueObjects.push(value);
		}		
		
		this.state = {
			values: valueObjects
		};
	}	
	
	/**
	 * Event Handler for if a checkbox value is changed.  Calls this.props.onValueChanged and returns selected values
	 * 
	 * @param {object} identifier - The name of the checkbox that was selected
	 * @memberof CheckboxGroup
	 */
	onValueChanged(identifier){
		var selectedValues = [];		
		var values = this.state.values;
		for(var x = 0; x < values.length; x++){
			var value = values[x];
			if(value.identifier === identifier){
				value.isSelected = !value.isSelected;
			}
			if(value.isSelected){
				selectedValues.push(value);
			}
		}
		this.setState({
			values: values
		});
		this.props.onValueChanged(selectedValues);
	}	
	
	/**
	 * Renders CheckboxGroup component
	 *
	 * @returns {object} - Form Group component with checkboxes inside 
	 * @memberof CheckboxGroup
	 */
	render(){
		let placement = this.props.labelPlacement;
		let muiPlacement;		
		
		/* Converts our placements to material-ui placements */
		if(placement === "right"){ muiPlacement = "end"; } 
		else if(placement === "left"){ muiPlacement = "start"; }
		else{ muiPlacement = placement; }		
		
		return <FormGroup row={this.props.alignment === 'horizontal' ? true : false}>
		{this.state.values.map(val => 
				<FormControlLabel
				key={val.identifier}
				label={<div>
					<span>{val.displayText}</span>
					<Tooltip text={val.helperText} />
					</div>} 
				value={val.identifier} 
				labelPlacement={muiPlacement}
				control={<Checkbox 
					data-testid={"checkbox"}
					id='checkbox'
					color={this.props.color} 
					disabled={val.isDisabled} 
					checked={val.isSelected} 
					onChange={this.onValueChanged.bind(this, val.identifier)} 
					name={val.identifier} 
				/>}
			/>
		)}
		</FormGroup >;
	}
}

/**
 * Default Props
 */
CheckboxGroup.defaultProps = {
	color: "default",
	alignment: 'vertical',
	labelPlacement: "right"
};

/**
 * Expected Props
 */
CheckboxGroup.propTypes = {
	value: PropTypes.arrayOf(PropTypes.any),
	hasError: PropTypes.bool,
	subText: PropTypes.string,
	alignment: PropTypes.oneOf(["vertical", "horizontal"]),
	color: PropTypes.string,
	labelPlacement: PropTypes.oneOf(["left", "top", "right", "bottom"]),
	variant: PropTypes.oneOf(["standard", "filled", "outlined"]),
	inputLabel: PropTypes.string,
	placeholder: PropTypes.string,
	onValueChanged: PropTypes.func.isRequired,
	hideSelectedValueInInput: PropTypes.bool,
	config: PropTypes.shape({
		isMultiple: PropTypes.bool,
		maxCount: PropTypes.number, // Also used by Text
		alternateResponseAllowed: PropTypes.bool,
		options: PropTypes.arrayOf(PropTypes.shape({
			identifier: PropTypes.string.isRequired,
			displayText: PropTypes.string.isRequired,
			helperText: PropTypes.string,
			isSelected: PropTypes.bool,
			isDisabled: PropTypes.bool
		})).isRequired
	})
};

export default CheckboxGroup;