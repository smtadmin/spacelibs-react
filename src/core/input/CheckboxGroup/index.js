/*
 * File: /src/core/input/CheckboxGroup/index.js
 * Version: 0.0.2
 * Project: @siliconmtn/spacelibs-react
 * Description: Index files are used to make exporting easier
 * File Created: Monday, 8th February 2021 1:27 pm
 * Author: tyler Gaffaney (tyler.gaffaney@siliconmtn.com)
 * -----
 * Last Modified: Thursday, 11th February 2021 1:57 pm
 * Modified By: tyler Gaffaney (tyler.gaffaney@siliconmtn.com>)
 * -----
 * Copyright 2021, Silicon Mountain Technologies, Inc.
*/

import CheckboxGroup from './CheckboxGroup';

export default CheckboxGroup;