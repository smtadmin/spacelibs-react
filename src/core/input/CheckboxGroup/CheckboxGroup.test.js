/*
 * File: CheckboxGroup.test
 * Version: 0.0.2
 * Project: @siliconmtn/spacelibs-react
 * Description: Testing suite for the CheckboxGroup functionality
 * File Created: Thursday, 11th February 2021 9:55 am
 * Author: Justin Jeffrey (justin.jeffrey@siliconmtn.com)
 * -----
 * Last Modified: Thursday, 11th February 2021 2:12 pm
 * Modified By: tyler Gaffaney (tyler.gaffaney@siliconmtn.com>)
 * -----
 * Copyright 2021, Silicon Mountain Technologies, Inc.
 */

import React from "react";
import {
  render,
  screen
} from "@testing-library/react";
import CheckboxGroup from "./CheckboxGroup";
import userEvent from '@testing-library/user-event';

const singleOptionNoValues = {
  alignment: "horizontal",
  number: 1,
  identifier: "1",
  type: "date",
  label: "Date Action Completed",
  isRequired: true,
  value: [],
  onValueChanged: () => { },
  config: {
    options: [
      {
        identifier: "A",
        displayText: "Option A",
        isSelected: undefined,
        isDisabled: undefined,
      }
    ],
  },
};

const twoOptionsOneSelectedValue = {
  number: 1,
  identifier: "1",
  type: "date",
  label: "Date Action Completed",
  isRequired: true,
  value: [
    {
      identifier: "B",
      displayText: "Option B",
      isDisabled: false,
    },
  ],
  onValueChanged: () => {},
  subText: "Error occurred",
  config: {
    options: [
      {
        identifier: "A",
        displayText: "Option A",
        isSelected: undefined,
        isDisabled: undefined,
      },
      {
        identifier: "B",
        displayText: "Option B",
        isDisabled: false,
      },
    ],
  },
};

const configWithUndefinedOptionValues = {
  number: 1,
  identifier: "1",
  type: "date",
  label: "Date Action Completed",
  isRequired: true,
  value: [
  ],
  onValueChanged: () => { },
  subText: "Error occurred",
  config: {
    options: [
      {
        identifier: "A",
        displayText: "Option A",
        isSelected: undefined,
        isDisabled: undefined,
      },
      {
        identifier: "B",
        displayText: "Option B",
        isDisabled: false,
      },
    ],
  },
};

/**
 * CheckboxGroup smoke test
 */
it("Renders without crashing", () => {
  const element = render(<CheckboxGroup {...twoOptionsOneSelectedValue} value={[]} />);
  expect(element).toBeTruthy();
});

/**
 * It renders a single values in the checkbox correctly
 */
it("Renders with single value", () => {
  const element = render(<CheckboxGroup {...singleOptionNoValues} />);
  expect(element).toMatchSnapshot();
});

/**
 * It renders multiple values in the checkbox correctly
 */
it("Renders with values", () => {
  const element = render(<CheckboxGroup {...twoOptionsOneSelectedValue} />);
  expect(element).toMatchSnapshot();
});

/**
 * It renders out a checkbox with the label in the correct place when the labelPlacement prop is set to left
 */
it("Renders correctly when labelPlacement is set to left", () => {
  const element = render(<CheckboxGroup {...singleOptionNoValues} labelPlacement="left" />);
  expect(element).toMatchSnapshot();
});

/**
 * It renders out a checkbox with the label in the correct place when the labelPlacement prop is set to top
 */
it("Renders correctly when labelPlacement is set to the top position", () => {  const element = render(<CheckboxGroup {...singleOptionNoValues} labelPlacement="top" />);
  expect(element).toMatchSnapshot();
});

/**
 * It correctly fires the callback when a user input is detected
 */
it("updates the correct field when a checkbox is selected", () => {
  const mockedFunction = jest.fn();  render(<CheckboxGroup {...configWithUndefinedOptionValues} onValueChanged={mockedFunction} />);
  const element = screen.getAllByTestId('checkbox')[0];
  const input = element.children[0].children[0];
  userEvent.click(input);
  expect(mockedFunction).toHaveBeenCalled();
});
