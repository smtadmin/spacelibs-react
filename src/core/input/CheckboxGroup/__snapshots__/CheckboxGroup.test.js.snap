// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly when labelPlacement is set to left 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiFormGroup-root MuiFormGroup-row"
      >
        <label
          class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-13 MuiCheckbox-root"
            data-testid="checkbox"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-16"
                data-indeterminate="false"
                id="checkbox"
                name="A"
                type="checkbox"
                value="A"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            <div>
              <span>
                Option A
              </span>
            </div>
          </span>
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiFormGroup-root MuiFormGroup-row"
    >
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-13 MuiCheckbox-root"
          data-testid="checkbox"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              class="PrivateSwitchBase-input-16"
              data-indeterminate="false"
              id="checkbox"
              name="A"
              type="checkbox"
              value="A"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          <div>
            <span>
              Option A
            </span>
          </div>
        </span>
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Renders correctly when labelPlacement is set to the top position 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiFormGroup-root MuiFormGroup-row"
      >
        <label
          class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementTop"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-17 MuiCheckbox-root"
            data-testid="checkbox"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-20"
                data-indeterminate="false"
                id="checkbox"
                name="A"
                type="checkbox"
                value="A"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            <div>
              <span>
                Option A
              </span>
            </div>
          </span>
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiFormGroup-root MuiFormGroup-row"
    >
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementTop"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-17 MuiCheckbox-root"
          data-testid="checkbox"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              class="PrivateSwitchBase-input-20"
              data-indeterminate="false"
              id="checkbox"
              name="A"
              type="checkbox"
              value="A"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          <div>
            <span>
              Option A
            </span>
          </div>
        </span>
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Renders with single value 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiFormGroup-root MuiFormGroup-row"
      >
        <label
          class="MuiFormControlLabel-root"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root"
            data-testid="checkbox"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-8"
                data-indeterminate="false"
                id="checkbox"
                name="A"
                type="checkbox"
                value="A"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            <div>
              <span>
                Option A
              </span>
            </div>
          </span>
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiFormGroup-root MuiFormGroup-row"
    >
      <label
        class="MuiFormControlLabel-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root"
          data-testid="checkbox"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              class="PrivateSwitchBase-input-8"
              data-indeterminate="false"
              id="checkbox"
              name="A"
              type="checkbox"
              value="A"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          <div>
            <span>
              Option A
            </span>
          </div>
        </span>
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Renders with values 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiFormGroup-root"
      >
        <label
          class="MuiFormControlLabel-root"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-9 MuiCheckbox-root"
            data-testid="checkbox"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-12"
                data-indeterminate="false"
                id="checkbox"
                name="A"
                type="checkbox"
                value="A"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            <div>
              <span>
                Option A
              </span>
            </div>
          </span>
        </label>
        <label
          class="MuiFormControlLabel-root"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-9 MuiCheckbox-root PrivateSwitchBase-checked-10 Mui-checked"
            data-testid="checkbox"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                checked=""
                class="PrivateSwitchBase-input-12"
                data-indeterminate="false"
                id="checkbox"
                name="B"
                type="checkbox"
                value="B"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            <div>
              <span>
                Option B
              </span>
            </div>
          </span>
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiFormGroup-root"
    >
      <label
        class="MuiFormControlLabel-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-9 MuiCheckbox-root"
          data-testid="checkbox"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              class="PrivateSwitchBase-input-12"
              data-indeterminate="false"
              id="checkbox"
              name="A"
              type="checkbox"
              value="A"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          <div>
            <span>
              Option A
            </span>
          </div>
        </span>
      </label>
      <label
        class="MuiFormControlLabel-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-9 MuiCheckbox-root PrivateSwitchBase-checked-10 Mui-checked"
          data-testid="checkbox"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input-12"
              data-indeterminate="false"
              id="checkbox"
              name="B"
              type="checkbox"
              value="B"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          <div>
            <span>
              Option B
            </span>
          </div>
        </span>
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
